### DISCUSSION TITLE
Some parts of core's extract_content function are not understood.

### DISCUSSION THREAD

**Discussion by jake-yi** [adbar/trafilatura#287] [Q&A]
Hello.

There's a part that I don't understand.

That is the 538th line of "https://github.com/adbar/trafilatura/blob/master/trafilatura/core.py"

```ptest = subtree.xpath(//p/text())```

In this part, it is marked as `//p//text()` so It's looking for the root's p-tag rather than the subtree's p-tag.

Isn't `./p//text()` the right approach?

Thank you.

**Comment by adbar** [comment]
Hi @jake-yi, I believe it doesn't make any difference because the subtree is used as root and not the tree but I may be wrong, I'm going to make tests.

**Comment by adbar** [comment]
```
>>> from lxml import etree
>>> tree = etree.fromstring('<html><body><div id="1"><p>ABC</p></div><div id="2"><p>DEF</p></div></body></html>')
>>> subtree = tree.xpath("//div")[0]
>>> subtree.xpath("//p/text()")
['ABC', 'DEF']
```

OK, my bad, thanks for your feedback!
